source/common/http/header_utility.cc

```diff
@@ -35,15 +37,17 @@ HeaderUtility::HeaderData::HeaderData(const envoy::api::v2::route::HeaderMatcher
     range_.set_start(config.range_match().start());
     range_.set_end(config.range_match().end());
     break;
+  case envoy::api::v2::route::HeaderMatcher::kPresentMatch:
+    header_match_type_ = HeaderMatchType::Present;
+    break;
   case envoy::api::v2::route::HeaderMatcher::HEADER_MATCH_SPECIFIER_NOT_SET:
     FALLTHRU;
   default:
     if (PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, regex, false)) {
       header_match_type_ = HeaderMatchType::Regex;
       regex_pattern_ = RegexUtil::parseRegex(config.value());
     } else {
-      header_match_type_ = HeaderMatchType::Value;
-      value_ = config.value();
+      header_match_type_ = HeaderMatchType::Present;
```

Comment:
To clarify my other comment, I think we want:

```
if (config.value().empty()) {
  // e.g. no HeaderMatchSpecifier was set (or an unknown one) and value is ""
  header_match_type_ = HeaderMatchType::Present;
} else {
  header_match_type_ = HeaderMatchType::Value;
  value_ = config.value();
}
```

Reply:
Thanks for the clarification. This makes sense to me.